protocol/app/app.go

```diff
@@ -234,6 +237,7 @@ type App struct {
 	ParamsKeeper     paramskeeper.Keeper
 	// IBC Keeper must be a pointer in the app, so we can SetRouter on it correctly
 	IBCKeeper             *ibckeeper.Keeper
+	EvidenceKeeper        evidencekeeper.Keeper
```

Comment:
I think you also need to add `x/evidence` to module basic [here](https://github.com/dydxprotocol/v4-chain/blob/6d30844950ba73e1d1003e11f320e1ee3529ca70/protocol/app/basic_manager/basic_manager.go#L49)